mods/parsoid.js

```diff
@@ -615,10 +611,18 @@ PSP.makeTransform = function(from, to) {
         return transform
         .then(function(res) {
             // Unwrap to the flat response format
-            var innerRes = res.body[to];
+            var innerRes = {};
+            if (to === 'wikitext') {
+                innerRes.body = res.body;
+                innerRes.headers = {
+                    'content-type': res.headers['content-type']
+                };
+            } else {
+                innerRes = res.body[to];
+            }
             innerRes.status = 200;
             // Handle bodyOnly flag
-            if (to === 'html' && req.body.bodyOnly) {
+            if (from === 'sections' && to === 'html' && req.body.bodyOnly) {
```

Comment:
See above: because the section extraction code adds a <body> around its result, which then has to be stripped off.  Parsoid no longer does this.  Help is requested to fix up the section extraction code, since I don't understand it well enough (and my daughter is going to be born within a matter of hours now).